lib/src/configuration.dart

```diff
@@ -70,20 +50,119 @@ class Configuration {
     return Directory.current.absolute.path;
   }
 
+  String? get fifoFilesFallbackPath;
+  String get path;
+  RealmSchema get schema;
+  int get schemaVersion;
+  List<int>? get encryptionKey;
+
+  @Deprecated('Use Configuration.local instead')
+  factory Configuration(
+    List<SchemaObject> schemaObjects, {
+    Function(Realm realm)? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+    bool disableFormatUpgrade,
+    bool isReadOnly,
+    bool Function(int totalSize, int usedSize)? shouldCompactCallback,
+  }) = LocalConfiguration;
+
+  factory Configuration.local(
+    List<SchemaObject> schemaObjects, {
+    Function(Realm realm)? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+    bool disableFormatUpgrade,
+    bool isReadOnly,
+    bool Function(int totalSize, int usedSize)? shouldCompactCallback,
+  }) = LocalConfiguration;
+
+  factory Configuration.inMemory(
+    List<SchemaObject> schemaObjects,
+    String identifier, {
+    Function(Realm realm)? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+  }) = InMemoryConfiguration;
+
+  factory Configuration.flexibleSync(
+    User user,
+    List<SchemaObject> schemaObjects, {
+    Function(Realm realm)? initialDataCallback,
```

Comment:
I was thinking it might make sense, when a new client syncs for the first time, that it adds some data to the realm about itself?

Reply:
This would probably be a bad idea since it can lead to unintended data duplication or a lot of conflicts. With local realms, we know that there's no data the first time a Realm is created - we don't have such guarantees with sync, so we should encourage people to do a check for the data after the realm is opened and only insert if necessary.

Reply:
If you want to do something on every first install? I expect `intialDataCallback` to be called the first time an app is opened (and opens the realm) after it is first installed - is that correct?

Reply:
I'll move to `LocalConfiguration` only